src/assets/js/protocol/protocol-lang-switcher.js

```diff
@@ -0,0 +1,63 @@
+// create namespace
+if (typeof Mozilla === 'undefined') {
+    var Mozilla = {};
+}
+
+(function() {
+    'use strict';
+
+    var LangSwitcher = {};
+
+    /**
+     * Returns URL pathname with preceded by a new page locale.
+     * Assumes first path immediately after hostname is the page locale.
+     * @param {Object} Location interface
+     * @param {String} Newly selected language code e.g. `de`
+     * @return {String} pathname e.g. `/de/firefox/`
+     */
+    LangSwitcher.switchPath = function(location, newLang) {
+        var parts = location.pathname.slice(1).split('/');
+        var currentLang = '/' + parts[0] + '/';
+
+        // check that first path is a valid lang code.
+        if (!/^(\/\w{2}-\w{2}\/|\/\w{2,3}\/)/.test(currentLang)) {
+            return false;
+        }
+
+        var urlpath = parts.slice(1).join('/');
+        return '/' + newLang + '/' + urlpath + location.search;
+    };
+
+    /**
+     * Redirect page to destination URL if valid
+     * @param {String} destination
+     */
+    LangSwitcher.doRedirect = function(destination) {
+        if (destination) {
+            window.location.href = destination;
+        }
+    };
+
+    /**
+     * Initialize footer lang switcher.
+     * @param {function} Custom callback for analytics.
+     */
+    LangSwitcher.init = function(callback) {
+        var language = document.getElementById('mzp-c-language-switcher-select');
```

Comment:
Since we're using an ID does that mean we can only allow one language switcher per page? If so we should mention that in the usage notes. Or if we want to allow multiples we'd need to rework this a bit. I could imagine someone wanting a language switcher at both the top and bottom of a page (the elements would still need unique IDs when the molecule is implemented but we could leave the values up to the implementor).

But I'm also fine with enforcing it as a singleton, which at least covers our immediate needs on mozorg. So let's just note that we currently only support one switcher per page and we can always revisit later (especially since the footer is likely getting overhauled soon anyway).

Reply:
I think it's pretty safe to assume that at one point, a site may want both a lang switcher in the header and footer of a page. My initial thought was let's wait to see if that requirement comes before changing the existing functionality that we've ported here from bedrock, but looking at the code I think this is a simple enough change to make. Let's do this now 👍